### PR TITLE
fix #2955 - upgrade zk authorization check

### LINKED ISSUE #2955: FATAL Error performing upgrade validating ZNode ACLs
I installed 2.0.1 using Uno, manually inserted some data and then shutdown Accumulo. I then changed my uno config to point to snapshot and did a `uno install accumulo --no-deps`. I got this error after starting Accumulo:
<pre>
2022-09-23T14:36:16,822 [upgrade.UpgradeCoordinator] ERROR: FATAL: Error performing upgrade
java.lang.RuntimeException: Upgrade Failed! Error validating ZNode ACLs. Check the log for specific failed paths, check ZooKeeper troubleshooting in user documentation for instructions on how to fix.
        at org.apache.accumulo.manager.upgrade.Upgrader9to10.validateACLs(Upgrader9to10.java:171) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.upgrade.Upgrader9to10.upgradeZookeeper(Upgrader9to10.java:135) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.upgrade.UpgradeCoordinator.upgradeZookeeper(UpgradeCoordinator.java:157) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.Manager.setManagerState(Manager.java:263) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.Manager.getManagerLock(Manager.java:1493) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.Manager.run(Manager.java:1039) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
</pre>

### PR DESCRIPTION
**Pull request by EdColeman**
Fixes https://github.com/apache/accumulo/issues/2955 by using ZooKeeper digest in ACL checks.  (Code was provided by @dlmarion)

Manually performed upgrade from 2.0.1 to 2.1-0-SNAPHOT without problems.

Original issue is https://github.com/apache/accumulo/issues/2890

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java [hunk 1]
```diff
@@ -162,6 +165,21 @@ private static String getFmtTime(final long epoch) {
     return fmt.format(timestamp);
   }
 
+  /**
+   * Get the ZooKeeper digest based on the instance secret that is used within ZooKeeper for
+   * authentication. This method is primary intended to be used to valid ZooKeeper ACLs. Use
```
Comment (dlmarion) [inline core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java:170]:
```suggestion
   * authentication. This method is primary intended to be used to validate ZooKeeper ACLs. Use
```

File: core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java [hunk 2]
```diff
@@ -162,6 +165,21 @@ private static String getFmtTime(final long epoch) {
...
+   * {@link #digestAuth(ZooKeeper, String)} to add authorizations to ZooKeeper.
+   */
+  public static Id getZkAuthId(final String secret) {
```
Comment (dlmarion) [inline core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java:173]:
```suggestion
  public static Id getZkDigestAuthId(final String secret) {
```

**Review by dlmarion — APPROVED** [review]
Looks good, couple of small nits.

File: core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java (hunk 1 above)
Comment (EdColeman) [reply core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooUtil.java]: Addressed with e24e4d593f

**Review by dlmarion — APPROVED** [review]

**Pull request merged**